File: Practice /JAVA/src/com/datastructurePractice/bubble_sort.java
package com.datastructurePractice;

import java.util.Arrays;

public class bubble_sort {
    public int[] array;
    public int count;

    //parametrized constructor
    public bubble_sort(int[] a){
        this.array = a;
    }

    //sorting algorithm starts here
    public void sort(){
        int n = array.length;
        for(int i = 0;i < n;i++){
            System.out.println(Arrays.toString(array));
            for(int j = 0;j < n - 1 -i;j++){
                if(array[j] > array[j+1]){
                    int temp = array[j];
                    array[j] = array[j+1];
                    array[j+1] = temp;
                }
            }
        }
    }
    //printing array here
    public void printArray(){
        System.out.println("The Array : " + Arrays.toString(array));
    }
}

File: C++/bubble_sort.cpp
#include <iostream>

using namespace std;

void swap(int* m,int* n){
  int temp = *m;
  *m = *n;
  *n = temp;
}

void bubble_sort(int array[],int n){
  for(int i = 0;i < n;i++){
    for(int j = 0;j < n-i-1;j++){
      if(array[j] > array[j+1]){
        swap(&array[j],&array[j+1]);
      }
    }
  }
}
void printArray(int array[],int n){
  cout << "Array : ";
  for(int i = 0; i < n;i++){
    cout << array[i];
  }
}
int main(){
  int a[] = {6,2,9,4,1};
  int n = sizeof(a) / (sizeof(a[0]));
  bubble_sort(a,n);
  printArray(a,n);
}
